File: lib/java/src/main/java/com/dianping/cat/configuration/client/entity/Domain.java
package com.dianping.cat.configuration.client.entity;

import com.dianping.cat.configuration.client.BaseEntity;
import com.dianping.cat.configuration.client.Constants;
import com.dianping.cat.configuration.client.IVisitor;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class Domain extends BaseEntity<Domain> {
    private String id;
    private String ip;
    private boolean enabled = true;
    private int maxMessageSize = 1000;

    public Domain() {
    }

    public Domain(String id) {
        this.id = id;
    }

    @Override
    public void accept(IVisitor visitor) {
        visitor.visitDomain(this);
    }

    @Override
    public void mergeAttributes(Domain other) {
        assertAttributeEquals(other, Constants.ENTITY_DOMAIN, Constants.ATTR_ID, id, other.getId());

        if (other.getIp() != null) {
            ip = other.getIp();
        }

        enabled = other.isEnabled();

        maxMessageSize = other.getMaxMessageSize();
    }

    public Domain setEnabled(boolean enabled) {
        this.enabled = enabled;
        return this;
    }

    public Domain setId(String id) {
        this.id = id;
        return this;
    }

    public Domain setIp(String ip) {
        this.ip = ip;
        return this;
    }

    public Domain setMaxMessageSize(int maxMessageSize) {
        this.maxMessageSize = maxMessageSize;
        return this;
    }

}

File: lib/csharp/src/CatClient/Configuration/Domain.cs
namespace Org.Unidal.Cat.Configuration
{
    /// <summary>
    ///   描述当前系统的情况
    /// </summary>
    public class Domain
    {
        private string _id = CatConstants.UNKNOWN_DOMAIN;
        private bool _mEnabled = true;
        private int _mMaxMessageSize = 1000;

        /// <summary>
        ///   当前系统的标识
        /// </summary>
        public string Id
        {
            get { return _id; }
            set { _id = value; }
        }

        ///// <summary>
        /////   当前系统的IP
        ///// </summary>
        //public string Ip { get; set; }

        /// <summary>
        ///   Cat日志是否开启，默认开启
        /// </summary>
        public bool Enabled
        {
            get { return _mEnabled; }
            set { _mEnabled = value; }
        }

        /// <summary>
        ///   当内存中的消息数量大于这个值时，会截短消息树，将截短的部分发送到服务端。
        /// </summary>
        public int MaxMessageSize
        {
            get { return _mMaxMessageSize;  }
            set { _mMaxMessageSize = value; }
        }
    }
}